src/data/polynomial/degree/erase_lead.lean

```diff
@@ -0,0 +1,173 @@
+/-
+Copyright (c) 2020 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import data.polynomial.degree.basic
+import data.polynomial.degree.trailing_degree
+
+/-!
+# Erase the leading term of a univariate polynomial
+
+## Definition
+
+* `erase_lead f`: the polynomial `f - leading term of f`
+
+erase_lead serves as the reduction step in an induction, shaving off one monomial from a polynomial.
+-/
+
+open polynomial finsupp finset
+
+namespace polynomial
+
+variables {R : Type*} [semiring R] {f : polynomial R}
+
+/-- erase_lead of a polynomial f is the polynomial obtained by
+subtracting from f the leading term of f. -/
+def erase_lead (f : polynomial R) : polynomial R :=
+ ⟨ f.support \ singleton f.nat_degree , λ a : ℕ , ite (a = f.nat_degree) 0 f.coeff a , λ a , begin
+  simp only [mem_sdiff, mem_support_iff, ne.def, mem_singleton],
+  split_ifs with h,
+  { simp only [not_and, pi.zero_apply, not_not, eq_self_iff_true, not_true, iff_false],
+    intros a_1,
+    assumption, },
+  { split,
+  { rw and_imp,
+    intros a1 a2,
+    assumption, },
+  { intros a1,
+    exact ⟨ a1 , h ⟩, }, },
+end ⟩
+
+namespace erase_lead
+
+lemma support (f : polynomial R) :
+ (f.erase_lead).support = f.support \ singleton f.nat_degree :=
+rfl
+
+@[simp] lemma coeff_nat_degree : f.erase_lead.coeff f.nat_degree = 0 :=
+begin
+  unfold erase_lead,
+  simp only [coeff_mk, if_true, pi.zero_apply, eq_self_iff_true],
+end
+
+@[simp] lemma coeff_eq_coeff_of_ne {a : ℕ} (h : a ≠ f.nat_degree) :
+ f.erase_lead.coeff a = f.coeff a :=
+begin
+  unfold erase_lead,
+  rw coeff_mk,
+  split_ifs,
+  { exfalso,
+    solve_by_elim, },
+  { refl, },
+end
+
+lemma sum_leading_C_mul_X_pow (f : polynomial R)
+ : f = f.erase_lead + (C f.leading_coeff) * X^f.nat_degree :=
```

Comment:
```suggestion
lemma erase_lead_add_C_mul_X_pow (f : polynomial R) :
  f.erase_lead + (C f.leading_coeff) * X^f.nat_degree = f :=
```

Reply:
This makes me think that it might be useful to have an inductive definition of polynomials, where you say that (0:polynomial R) is a polynomial and that if (P : polynomial R), (r : R), (n : \N), then P + (C r) * X^n : polynomial R.